### PR TITLE
docs(cli-ui): add more notes for local development

### PR DESCRIPTION
**Pull request by Kocal**
This should _improve_ actual README with more user-friendly notes for starting local development and explaining how to run tests in local aswell.

### CODE REVIEW

File: packages/@vue/cli-ui/README.md [hunk 1]
```diff
@@ -1,15 +1,43 @@
...
+```bash
+cd ../cli-ui-addon-webpack
+yarn build
+cd -
+```
+
+Then you should start a local Apollo server:
+
+```
+yarn run apollo
```
Comment (Akryum) [inline packages/@vue/cli-ui/README.md]: This needs to be run in the `cli-ui` folder (not `cli-ui-addon-webpack`).

File: packages/@vue/cli-ui/README.md [hunk 2]
```diff
@@ -1,15 +1,43 @@
...
+in order to download all required dependencies of vue-cli's packages.
+ 
+Just after, you should build `@vue/cli-ui-addon-webpack` by running:
+```bash
+cd ../cli-ui-addon-webpack
+yarn build
+cd -
+```
+
+Then you should start a local Apollo server:
```
Comment (Akryum) [inline packages/@vue/cli-ui/README.md]: `start the ui server (based on Apollo)`

File: packages/@vue/cli-ui/README.md [hunk 3]
```diff
@@ -1,15 +1,43 @@
...
+
+```
+yarn run apollo
+```
+
+And then in another terminal, you should serve cli-ui:
```
Comment (Akryum) [inline packages/@vue/cli-ui/README.md]: `you should serve the ui web app`

File: packages/@vue/cli-ui/README.md [hunk 4]
```diff
@@ -1,15 +1,43 @@
...
+### Local development
+
+Before starting a local cli-ui instance, 
+consider following the [contributing guide](https://github.com/vuejs/vue-cli/blob/dev/.github/CONTRIBUTING.md) 
+in order to download all required dependencies of vue-cli's packages.
+ 
+Just after, you should build `@vue/cli-ui-addon-webpack` by running:
+```bash
+cd ../cli-ui-addon-webpack
+yarn build
```
Comment (Akryum) [inline packages/@vue/cli-ui/README.md]: Instead: run `yarn serve` in a 3rd terminal.

File: packages/@vue/cli-ui/README.md [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
+#### Testing
+
+Before running E2E tests, you should start a new local testing apollo server with the following command:
 
 ```
-yarn run apollo
+yarn run test:e2e
+```
+
+Then in another terminal:
```
Comment (Akryum) [inline packages/@vue/cli-ui/README.md]: `yarn run test:e2e` should be enough. It both starts the apollo server and the vue-cli e2e service (`yarn run test:e2e:dev`).

File: packages/@vue/cli-ui/README.md (hunk 1 above)
Comment (Kocal) [reply packages/@vue/cli-ui/README.md]: I though `cd -` (like it's done in https://github.com/vuejs/vue-cli/blob/dev/.github/CONTRIBUTING.md) was enough. :thinking: 

Anyway I'm gonna change it to be more clear

File: packages/@vue/cli-ui/README.md (hunk 5 above)
Comment (Kocal) [reply packages/@vue/cli-ui/README.md]: It seems not.

When running `yarn run test:e2e` alone, it just starts an Apollo server and waits (for nothing).
I have to run both command `yarn run test:e2e` and `yarn run test:e2e:dev` to make tests work.

Maybe this is a bug?

Comment (Akryum) [reply packages/@vue/cli-ui/README.md]: Make sure `http://localhost:4040` works. It should start `test:e2e:dev` script as soon as `http://localhost:4040` responds.

File: packages/@vue/cli-ui/README.md (hunk 4 above)
Comment (Kocal) [reply packages/@vue/cli-ui/README.md]: Hum, I think I will revert to `yarn build`. Using `yarn serve` give me this error during tests:

```
Error: Cannot find module '@vue/cli-ui-addon-webpack/dist'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:548:15)
    at Function.resolve (internal/modules/cjs/helpers.js:18:19)
    at serve (/home/kocal/Dev/vue-cli/packages/@vue/cli-ui/apollo-server/connectors/client-addons.js:56:34)
    at Layer.handle [as handle_request] (/home/kocal/Dev/vue-cli/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/kocal/Dev/vue-cli/node_modules/express/lib/router/index.js:317:13)
    at /home/kocal/Dev/vue-cli/node_modules/express/lib/router/index.js:284:7
    at param (/home/kocal/Dev/vue-cli/node_modules/express/lib/router/index.js:354:14)
    at param (/home/kocal/Dev/vue-cli/node_modules/express/lib/router/index.js:365:14)
    at param (/home/kocal/Dev/vue-cli/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/home/kocal/Dev/vue-cli/node_modules/express/lib/router/index.js:410:3)
```

File: packages/@vue/cli-ui/README.md (hunk 5 above)
Comment (Kocal) [reply packages/@vue/cli-ui/README.md]: Yep it works, it show me a GraphQL Playground webapp.

This is my output:

```
➜  cli-ui git:(chore/docs-cli-ui) yarn run test:e2e                  
yarn run v1.7.0
$ yarn run test:clear && start-server-and-test apollo:run:test http://localhost:4040 test:e2e:dev
$ rimraf ../../test/cli-ui-test && rimraf ./live-test
starting server using command "npm run apollo:run:test"
and when url "http://localhost:4040" is responding
running tests using command "test:e2e:dev"

> @vue/cli-ui@3.0.1 apollo:run:test /home/kocal/Dev/vue-cli/packages/@vue/cli-ui
> cross-env VUE_CLI_UI_TEST=true VUE_APP_GRAPHQL_PORT=4040 VUE_APP_CLI_UI_URL=ws://localhost:4040/graphql VUE_CLI_IPC=vue-cli-test vue-cli-service apollo:watch

$ cross-env VUE_CLI_DEBUG=true VUE_CLI_IPC=vue-cli-dev vue-cli-service apollo:run --delay
✔️  GraphQL Server is running on http://localhost:4040/graphql
```

File: packages/@vue/cli-ui/README.md (hunk 4 above)
Comment (Akryum) [reply packages/@vue/cli-ui/README.md]: For the tests you need to build the webpack addon at least once.

File: packages/@vue/cli-ui/README.md (hunk 5 above)
Comment (Akryum) [reply packages/@vue/cli-ui/README.md]: So there is something strange here

Comment (Kocal) [reply packages/@vue/cli-ui/README.md]: I'm gonna open a new issue and apply your changes

File: packages/@vue/cli-ui/README.md (hunk 4 above)
Comment (Kocal) [reply packages/@vue/cli-ui/README.md]: So what's the point for using `yarn serve`, since it doesn't output anything in `dist` folder?

Comment (Akryum) [reply packages/@vue/cli-ui/README.md]: It's serving the hot-reloaded webpack dashboard components on localhost which are then loaded in the UI when run in dev mode.

Comment (Kocal) [reply packages/@vue/cli-ui/README.md]: Updated.

**Pull request merged**